solarized-definitions.el

```diff
@@ -237,30 +237,45 @@ the \"Gen RGB\" column in solarized-definitions.el to improve them further."
                     (diff-changed ((t (,@fmt-revr ,@fg-yellow))))
                     (diff-removed ((t (,@fmt-revr ,@fg-red))))
                     (diff-refine-change
-                     ((t (,@fmt-revr ,@fg-blue ,@bg-back))))))
+                     ((t (,@fmt-revr :foreground "black" ,@bg-base01))))))
                  (low
                   `((diff-added ((t (,@fmt-undr ,@fg-green))))
                     (diff-changed ((t (,@fmt-undr ,@fg-yellow))))
                     (diff-removed ((t (,@fmt-bold ,@fg-red))))
                     (diff-refine-change
-                     ((t (,@fmt-undr ,@fg-blue ,@bg-back))))))
+                     ((t (,@fmt-undr :foreground "black" ,@bg-back))))))
                  (normal
                   (if window-system
                       `((diff-added ((t (,@fmt-bold ,@fg-green))))
                         (diff-changed ((t (,@fmt-bold ,@fg-yellow))))
                         (diff-removed ((t (,@fmt-bold ,@fg-red))))
                         (diff-refine-change
-                         ((t (,@fmt-bold ,@fg-blue ,@bg-back)))))
+                         ((t (,@fmt-bold :foreground "black" ,@bg-base01)))))
                     `((diff-added ((t (,@fg-green))))
                       (diff-changed ((t (,@fg-yellow))))
                       (diff-removed ((t (,@fg-red))))
-                      (diff-refine-change ((t (,@fg-blue ,@bg-back))))))))
+                      (diff-refine-change ((t (:foreground "black" ,@bg-base01))))))))
              (diff-file-header ((t (,@bg-back))))
              (diff-header ((t (,@fg-base1 ,@bg-back))))
              ;; IDO
              (ido-only-match ((t (,@fg-green))))
              (ido-subdir ((t (,@fg-blue))))
              (ido-first-match ((t (,@fmt-bold ,@fg-green))))
+             ;; helm
+             (helm-source-header ((t (,@fg-base01 ,@bg-base02 :height 1.2 ,@fmt-none))))
+             (helm-ff-directory ((t (,@fg-cyan :background nil))))
+             (helm-candidate-number ((t (,@fmt-undr ,@fg-blue :background nil))))
+             (helm-ff-file ((t (:inherit default))))
+             (helm-ff-executable ((t (,@fg-green))))
+             (helm-selection ((t (,@bg-blue ,@fg-base02 ,@fmt-none))))
+             ;; iedit
+             (iedit-occurrence ((t (,@fg-base3 ,@bg-violet))))
+             ;; eldoc
+             (eldoc-highlight-function-argument ((t (,@fmt-undr ,@fg-cyan))))
+             ;; mmm-mode
+             (mmm-code-submode-face ((t (:background nil ,@fmt-ital))))
+             (mmm-output-submode-face ((t (:background nil ,@fmt-undi))))
+             (mmm-comment-submode-face ((t (:inherit font-lock-comment-face))))
```

Comment:
I would probably be likely to accept these additions, with some modifications:
* `:height` doesn’t get set in Solarized
* `:background nil` can now be done as `,@bg-none`